jib-core/src/test/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactoryTest.java

```diff
@@ -63,7 +67,12 @@ private static DockerCredentialHelperFactory getTestFactory(
     };
   }
 
-  @Mock private JibLogger mockJibLogger;
+  private StringBuilder logMessages = new StringBuilder();
+  private EventEmitter eventEmitter =
+      new DefaultEventEmitter(
+          new EventHandlers()
+              .add(JibEventType.LOGGING, logEvent -> logMessages.append(logEvent.getMessage())));
```

Comment:
I admittedly did not try it. I was going off of a different test that checked log messages and it did it this way.

Reply:
It looks like it'll work if I add an `equals` to `LogEvent`. Which I think is worth it, since it makes the tests a lot more straightforward. I'll just make the `equals` visible for testing.